core/types/receipt.go

```diff
@@ -610,11 +610,17 @@ func (rs Receipts) DeriveFields(config *params.ChainConfig, hash common.Hash, nu
 		rs[i].Type = txs[i].Type()
 		rs[i].TxHash = txs[i].Hash()
 
-		// The post transition CeloDynamicFeeV2Txs set the baseFee in the receipt
-		if rs[i].BaseFee == nil {
-			rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), baseFee)
-		} else {
-			rs[i].EffectiveGasPrice = txs[i].inner.effectiveGasPrice(new(big.Int), rs[i].BaseFee)
+		// Pre-gingerbred the base fee was stored in state, but we don't try to recover it here, since A) we don't have
```

Comment:
Could we store it in the client somewhere. I guess it was usually constant for a range, so this wouldn't be horribly inefficient.

Reply:
We could hardcode it like the gas limit, but we could also fetch it from the block receipts, which might be a bit nicer. But I'd consider either to be outside the scope of this PR.